casper/src/main/rholang/MakeMint.rho

```diff
@@ -47,18 +49,27 @@ in {
                 }
               }
             } |
-            contract thisPurse(@"deposit", @amount, @src, success) = {
-              new result in {
-                @src!(*decr, amount, *result) | //call src decr function.
-                for(@decrSuccess <- result) {
-                  if (decrSuccess) {
-                    balance!("add", amount, *success) // add transferred amount to this purse
+
+            contract thisPurse(@"deposit", @amount, @src, successCh) = {
+              new ch in {
+                balance!("add", amount, *ch) |
```

Comment:
NonNegativeNumber already checks for overflow. I don't see why this change is needed.

Reply:
oops; now I see; never mind...